controller/scheduler/scheduler.go

```diff
@@ -722,51 +734,64 @@ func (s *Scheduler) changeFormation(ef *ct.ExpandedFormation) (*Formation, error
 		}
 	}
 
-	f := s.formations.Get(ef.App.ID, ef.Release.ID)
+	f = s.formations.Get(ef.App.ID, ef.Release.ID)
 	if f == nil {
 		log.Info("adding new formation", "processes", ef.Processes)
-		f = s.formations.Add(NewFormation(ef))
+		f = s.formations.Add(NewFormation(ef, s.HandleRectify))
 	} else {
-		if !f.GetProcesses().Equals(ef.Processes) {
+		if f.GetProcesses().Equals(ef.Processes) {
+			return f, nil
+		} else {
 			log.Info("updating processes of existing formation", "processes", ef.Processes)
 			f.Processes = ef.Processes
 		}
 	}
+	s.triggerRectify(f.key())
 	return f, nil
 }
 
+func (s *Scheduler) triggerRectify(key utils.FormationKey) {
+	logger.Info("triggering rectify", "key", key)
+	s.formations.TriggerRectify(key)
```

Comment:
Thinking about it, another option is to have a buffer wrapper around the rectify channel, and have `s.rectify` receive a "batch" of formations, e.g.:

```go
batch := map[utils.FormationKey]struct{}
var out chan map[utils.FormationKey]struct{}
for {
  select {
  case key := <-in:
    // add key to batch
    batch[key] = struct{}{}
    // switch `out` so we try and send the batch on next loop
    out = s.rectify
  case out <- batch:
    // batch sent, reset
    batch = map[utils.FormationKey]struct{}
    out = nil
  }
}
```